File: mobile-app/lib/models/store.dart
class Store {
  final int id;
  final String name;
  final String phone;
  final String? logo;
  final String slug;
  final bool isActive;
  final int viewCount;
  final String storeUrl;
  final int productCount;
  final DateTime createdAt;
  final DateTime updatedAt;

  Store({
    required this.id,
    required this.name,
    required this.phone,
    this.logo,
    required this.slug,
    required this.isActive,
    required this.viewCount,
    required this.storeUrl,
    required this.productCount,
    required this.createdAt,
    required this.updatedAt,
  });

  factory Store.fromJson(Map<String, dynamic> json) {
    return Store(
      id: int.parse(json['id'].toString()),
      name: json['name'],
      phone: json['phone'],
      logo: json['logo'],
      slug: json['slug'],
      isActive: json['is_active'] == 1 || json['is_active'] == true,
      viewCount: int.parse((json['view_count'] ?? 0).toString()),
      storeUrl: json['store_url'] ?? '',
      productCount: int.parse((json['product_count'] ?? 0).toString()),
      createdAt: DateTime.parse(json['created_at']),
      updatedAt: DateTime.parse(json['updated_at']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'phone': phone,
      'logo': logo,
      'slug': slug,
      'is_active': isActive,
      'view_count': viewCount,
      'store_url': storeUrl,
      'product_count': productCount,
      'created_at': createdAt.toIso8601String(),
      'updated_at': updatedAt.toIso8601String(),
    };
  }

  Store copyWith({
    int? id,
    String? name,
    String? phone,
    String? logo,
    String? slug,
    bool? isActive,
    int? viewCount,
    String? storeUrl,
    int? productCount,
    DateTime? createdAt,
    DateTime? updatedAt,
  }) {
    return Store(
      id: id ?? this.id,
      name: name ?? this.name,
      phone: phone ?? this.phone,
      logo: logo ?? this.logo,
      slug: slug ?? this.slug,
      isActive: isActive ?? this.isActive,
      viewCount: viewCount ?? this.viewCount,
      storeUrl: storeUrl ?? this.storeUrl,
      productCount: productCount ?? this.productCount,
      createdAt: createdAt ?? this.createdAt,
      updatedAt: updatedAt ?? this.updatedAt,
    );
  }
}

File: backend/app/Models/Store.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;
use Illuminate\Support\Str;

class Store extends Model
{
    use HasFactory, SoftDeletes;

    protected $fillable = [
        'name',
        'phone',
        'logo',
        'slug',
        'is_active',
        'view_count',
    ];

    protected $casts = [
        'is_active' => 'boolean',
        'view_count' => 'integer',
    ];

    protected $appends = ['store_url', 'product_count'];

    /**
     * Boot the model.
     */
    protected static function boot()
    {
        parent::boot();

        static::creating(function ($store) {
            if (empty($store->slug)) {
                $store->slug = Str::slug($store->name) . '-' . Str::random(6);
            }
        });
    }

    /**
     * Get the products for the store.
     */
    public function products()
    {
        return $this->hasMany(Product::class);
    }

    /**
     * Get the analytics events for the store.
     */
    public function analyticsEvents()
    {
        return $this->hasMany(AnalyticsEvent::class);
    }

    /**
     * Get the store URL attribute.
     */
    public function getStoreUrlAttribute()
    {
        return config('app.storefront_url') . '/store/' . $this->slug;
    }

    /**
     * Get the product count attribute.
     */
    public function getProductCountAttribute()
    {
        return $this->products()->count();
    }

    /**
     * Increment view count.
     */
    public function incrementViewCount()
    {
        $this->increment('view_count');
    }

    /**
     * Get active products.
     */
    public function activeProducts()
    {
        return $this->products()->where('is_active', true);
    }
}
